onnx/defs/controlflow/defs.cc

```diff
@@ -494,18 +494,25 @@ C-style code:
       const int max_trip_count = 10; // Analogous to input M
       int user_defined_vals[]; // Imagine this is resizable
       /* End implicitly-defined code */
+      int b_out = b;
+      bool keepgoing_out = keepgoing;
```

Comment:
My personal preference would be to put the above two lines (assignments to b_out and keepgoing_out) _after_ the loop. There needs to be only one "loop-carried" version of each variable, and those seem to be "b" and "keepgoing". Technically, both have the same effect, so this is just a minor nitpick.